cloud/amazon/ec2_vpc.py

```diff
@@ -227,6 +227,76 @@ def find_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
     return (found_vpc)
 
+#Check if the route table has all routes as in given list
+def routes_match(rt_list=None, rt=None, igw=None):
+    local_routes = []
+    remote_routes = []
+    for route in rt_list:
+        route_kwargs = {}
+        if route['gw'] == 'igw':
+            route_kwargs['gateway_id'] = igw.id
+            route_kwargs['instance_id'] = None
+            route_kwargs['state'] = 'active'
+        elif route['gw'].startswith('i-'):
+            route_kwargs['instance_id'] = route['gw']
+            route_kwargs['gateway_id'] = None
+            route_kwargs['state'] = 'active'
+        else:
+            route_kwargs['gateway_id'] = route['gw']
+            route_kwargs['instance_id'] = None
+            route_kwargs['state'] = 'active'
+        route_kwargs['destination_cidr_block'] = route['dest']
+        local_routes.append(route_kwargs)
+    for j in rt.routes:
+        remote_routes.append(j.__dict__)
+    match = []
+    for i in local_routes:
+        change = "false"
+        for j in remote_routes:
+            if set(i.items()).issubset(set(j.items())):
+                change = "true"
+        match.append(change)
+    if 'false' in match:
+        return False
+    else:
+        return True
+
+#Check if the remote routes match the local routes
+def rtb_match(route_tables=None, vpc_conn=None, module=None, vpc=None, igw=None):
```

Comment:
Sorry for not giving all my remarks at the same time, but you may want to rename this function to reflect that it returns true if the rtb has changed. rtb_match implies the opposite behavior. (you could also swap true/false in the function itself).

The prefered way to add documantation of a function is like this:
    """
        Checks if the remote routes match the local routes.
        
        route_tables : description of the route_tables parameter
        vpc_conn : description of the vpc_conn parameter
        module : description of the module parameter
    vpc_conn: authenticated VPCConnection connection object
    Returns:
        A dictionary with information
        about the VPC and subnets that were launched
    """